pkg/kubelet/pod_workers.go

```diff
@@ -131,6 +132,15 @@ func (p *podWorkers) UpdatePod(pod *api.Pod, mirrorPod *api.Pod, updateComplete
 		// the channel is empty, so buffer of size 1 is enough.
 		podUpdates = make(chan workUpdate, 1)
 		p.podUpdates[uid] = podUpdates
+		if firstSeenTimeStr, ok := pod.Annotations[ConfigFirstSeenAnnotationKey]; !ok {
+			glog.Errorf("First seen time not recorded for pod %s", pod.UID)
+		} else {
+			if firstSeenTime, err := time.Parse(time.RFC3339, firstSeenTimeStr); err != nil {
+				glog.Errorf("First seen time for %s cannot be parsed: %v", pod.UID, err)
+			} else {
+				metrics.StartPodWorkerLatency.Observe(metrics.SinceInMicroseconds(firstSeenTime))
```

Comment:
It would actually be more accurate if you recorded this time when syncPod starts instead of when we create the goroutine.